examples/app/surface3d/tsconfig.json

```diff
@@ -0,0 +1,35 @@
+{
+  "compilerOptions": {
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "noImplicitReturns": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "strictNullChecks": true,
+    "strictBindCallApply": false,
+    "strictFunctionTypes": false,
+    "strictPropertyInitialization": false,
+    "alwaysStrict": true,
+    "noErrorTruncation": true,
+    "noEmitOnError": false,
+    "declaration": true,
+    "sourceMap": true,
+    "importHelpers": false,
+    "experimentalDecorators": true,
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "target": "ES5",
+    "lib": ["es2015", "dom"],
+    "baseUrl": ".",
+    "outDir": "./dist/lib",
+    "paths": {
+      "@bokehjs/*": [
+        "./node_modules/bokehjs/build/js/lib/*",
+        "./node_modules/bokehjs/build/js/types/*"
+      ]
+    }
+  },
+  "include": ["./**/*.ts"]
+}
```

Comment:
I'm really concerned about this. Creating extensions is already a hurdle for almost all users. Adding all this much more will just basically make it impossible. There's no way most users can be expected to be successful with this, especially without documentation support.

Reply:
Not six but two. The minimum setup is an `index.ts` and a `bokeh.ext.json`, which is an indicator that a directory is a bokeh extension and in future will provide extension's configuration. Having `index.ts` can be made configurable in future. Other files, including `package.json` and `tsconfig.json` are optional. If you don't need dependencies, there is no need for `package.json`. `package-lock.json` is a side effect of how npm works, and if you accept using dependencies, you have to accept this (or one can disable generating this file by npm, though this would require allowing to pass options to npm). If `tsconfig.json` isn't provided, a default one will be used provided by the compiler, but this will make any IDE refuse to work with your `*.ts` files (no auto-completion, no error checking, etc.).

This work is about allowing to create a workflow for developing extensions, especially large extensions like e.g. `panel`. Complexity is inevitable and exposing everything necessary it the right way to do. From here we can look into how we can improve this setup for small, possibly inline extensions.